File: leetsync/2610-closest-prime-numbers-in-range/closest-prime-numbers-in-range.go
func closestPrimes(left int, right int) []int {
	isPrime := sieveHelper(right)
	primes := []int{}

	for num := left; num <= right; num++ {
		if isPrime[num] {
			primes = append(primes, num)
		}
	}
	if len(primes) < 2 {
		return []int{-1, -1}
	}
	minGap := int(^uint(0) >> 1)
	result := []int{-1, -1}
	for i := 1; i < len(primes); i++ {
		gap := primes[i] - primes[i-1]
		if minGap > gap {
			minGap = gap
			result[0] = primes[i-1]
			result[1] = primes[i]
		}
	}
	return result
}

func sieveHelper(right int) []bool {
	sieve := make([]bool, right+1)
	for i := range sieve {
		sieve[i] = true
	}
	sieve[0], sieve[1] = false, false

	for i := 2; i*i <= right; i++ {
		if sieve[i] {
			for j := i * i; j <= right; j += i {
				sieve[j] = false
			}
		}
	}
	return sieve
}

File: leetsync/2610-closest-prime-numbers-in-range/closest-prime-numbers-in-range.java
class Solution {
    public boolean[] sieveHelper(int right) {
        boolean[] isPrime = new boolean[right + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = false;
        isPrime[1] = false;
        for (int i = 2; i * i <= right; i++) {
            if (isPrime[i]) {
                for (int j = 2; i * j <= right; j++) {
                    isPrime[i * j] = false;
                }
            }
        }
        return isPrime;
    }

    public int[] closestPrimes(int left, int right) {
        boolean[] isPrime = sieveHelper(right);
        ArrayList<Integer> primes = new ArrayList<>();
        for (int num = left; num <= right; num++) {
            if (isPrime[num]) {
                primes.add(num);
            }
        }
        int minDiff = Integer.MAX_VALUE;
        int[] result = { -1, -1 };

        for (int i = 1; i < primes.size(); i++) {
            int diff = primes.get(i) - primes.get(i - 1);
            if (diff < minDiff) {
                minDiff = diff;
                result[0] = primes.get(i - 1);
                result[1] = primes.get(i);
            }
        }

        return result;
    }
}